### PR TITLE
Move failing NBitcoin RPC client tests into integration tests

### LINKED ISSUE #868: [Tests] NBitcoin tests that currently fail
These ones fail all the time (are from RPC Client Tests and REST Client Tests):

- CanGetBlockHeader
- CanGetRawMemPool
- CanGetTransaction
- CanGetTxOutAsyncFromRPC
- CanGetTxOutFromRPC
- CanGetUTXOsMempool
- CanUseAsyncRPC
- EstimateFeeRate
- TestFundRawTransaction

These ones I was able to get to pass by running alone but failed originally when running with full batch:

- CheckBlockProofOfStake
- IndexTheFullChain
- RawTransactionIsConformsToRPC

### PR DESCRIPTION
**Pull request by mikedennis**
There are several nbitcoin RPC client tests that are currently failing due to the peer refactoring. This PR moves those tests into integration test project so they can now use the new peer.

See #868 

Status of integration/nbitcoin tests for this PR can be seen here https://ci.appveyor.com/project/mikedennis/stratisci/branch/mikedennis

This PR also generalizes the RPCTestFixture class so it can be used for both Bitcoin and Stratis networks. The Bitcoin RPC tests are divided between tests that can can make use of the testfixture and others that require their own setup for each individual test.

### CODE REVIEW

**Comment by Aprogiena** [comment]
Careful with PR description keywords. `Fix X` or `Fixes X` is recognized pattern that will close issue X. Since you fix it only partially, you should avoid that.

**Comment by bokobza** [comment]
@mikedennis is this ready for review?

**Comment by mikedennis** [comment]
@bokobza Yes it is. There is more tests to fix in future but these are all the RPCClient ones.

**Comment by fassadlr** [comment]
@mikedennis so the 4 integration tests that are failing in your AppVeyor build is unrelated? They seem be RPC related.

**Comment by mikedennis** [comment]
@fassadlr Those integration tests are in ApiTests and they fail quite often for me in Appveyor but I can run them standalone. I don't think related to my change but something to follow up on.

**Comment by fassadlr** [comment]
@mikedennis I want to know whether the tests in `RpcBitcoinImmutableTests` pass in your AppVeyor integration build? I can't see them though. Sorry just a bit confused :(

**Comment by mikedennis** [comment]
Sorry @fassadlr I guess I put the wrong link to the appveyor in here. This one should show the test results for this branch https://ci.appveyor.com/project/mikedennis/stratisci/build/1.0.2.40

File: src/Stratis.Bitcoin.IntegrationTests/RPC/RpcBitcoinMutableTests.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            {
+                CoreNode nodeA = builder.CreateNode();
+                CoreNode nodeB = builder.CreateNode();
+                builder.StartAll();
+                RPCClient rpc = nodeA.CreateRPCClient();
+                rpc.RemoveNode(nodeA.Endpoint);
+                rpc.AddNode(nodeB.Endpoint);
+
+                AddedNodeInfo[] info = null;
+                WaitAssert(() =>
```
Comment (mikedennis) [inline src/Stratis.Bitcoin.IntegrationTests/RPC/RpcBitcoinMutableTests.cs]: I wonder whether I might be able to use this TestHelper.WaitLoop() instead?

Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests/RPC/RpcBitcoinMutableTests.cs]: Ah yes @mikedennis totally. Use the helper ;)

**Comment by fassadlr** [comment]
Thanks @mikedennis I have checked the tests that you have moved and they are all passing.

**Pull request merged**